### PR TITLE
Expose Context, EditorWatchdog, and ContextWatchdog as static editor properties

### LINKED ISSUE #13707: Expose `Context`, `EditorWatchdog`, and `ContextWatchdog` as static editor properties.
## 📝 Provide a description of the improvement

To simplify integrations and consistency with editor builds, we should expose `Context`, `EditorWatchdog`, and `ContextWatchdog` as static editor properties. Expose them on editors (not on the base `Editor` class) so we won't have a dependency from the `core` package to `watchdog`.

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Other (editor-*): Exposed `Context`, `EditorWatchdog`, and `ContextWatchdog` as static editor properties. Closes #13707.

MINOR BREAKING CHANGE: The `Watchdog` package is no longer available as a standalone [DLL build](https://ckeditor.com/docs/ckeditor5/latest/installation/advanced/alternative-setups/dll-builds.html). It is now included in the main `ckeditor5-dll.js` build. See #13707.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-editor-balloon/src/ballooneditor.ts [hunk 1]
```diff
@@ -237,6 +240,27 @@ export default class BalloonEditor extends DataApiMixin( ElementApiMixin( Editor
...
+	 * The {@link module:core/context~Context} class.
+	 *
+	 * Exposed as static editor field for easier access in editor builds.
+	 */
+	public static Context = Context;
+
+	/**
+	 * The {@link module:watchdog/editorwatchdog~EditorWatchdog} class.
+	 *
+	 *  Exposed as static editor field for easier access in editor builds.
```
Comment (Reinmar) [inline packages/ckeditor5-editor-balloon/src/ballooneditor.ts:254]:
```suggestion
	 * Exposed as static editor field for easier access in editor builds.
```

File: packages/ckeditor5-editor-balloon/src/ballooneditor.ts [hunk 2]
```diff
@@ -237,6 +240,27 @@ export default class BalloonEditor extends DataApiMixin( ElementApiMixin( Editor
...
+	 */
+	public static EditorWatchdog = EditorWatchdog;
+
+	/**
+	 * The {@link module:watchdog/contextwatchdog~ContextWatchdog} class.
+	 *
+	 *  Exposed as static editor field for easier access in editor builds.
```
Comment (Reinmar) [inline packages/ckeditor5-editor-balloon/src/ballooneditor.ts:261]:
```suggestion
	 * Exposed as static editor field for easier access in editor builds.
```

**Comment by Reinmar** [comment]
@pomek I think we should consider adding the watchdog to the core DLL. It's 600SLOC, so not much. And if we expect to use it as a dependency of editor packages, it'd be cleaner if it was imported there via `ckeditor5` instead of becoming a new dependency of those editor packages.

Looking at the arch diagram, watchdog would land for me in the "feature utils", if we agree that an editor creator is a "feature".



PS. As a followup we'd need to update the ESLint rule to expect watchdog to be

**Review by pomek — APPROVED** [review]
LGTM.

**Comment by pomek** [comment]
We talked F2F, and there are my proposals:

* Keep the watchdog DLL build (even empty) not to break existing integrations (removing the build results in error 404).
* Add (manually) a `console.warn()` into the build that we will remove in the next major update (`38.0.0`). Also, we can link to an issue where we describe what to do.

**Review by arkflpc — APPROVED** [review]

**Pull request merged**